### PR TITLE
Always select a diagonal for ArcTween even if begin and end rects are…

### LINKED ISSUE #5086: Gallery, Pesto: White space after tapping Home in pesto drawer
Open gallery, tap pesto, open drawer, tap home.

Expected: drawer transitions out and user is looking at a list of recipes (just like on initial load).

Actual: White space between app bar and recipe:

screen shot 2016-07-27 at 11 41 51 am

### PR DESCRIPTION
**Pull request by jason-simmons**
… equal

Fixes https://github.com/flutter/flutter/issues/5086

### CODE REVIEW

File: packages/flutter/lib/src/material/arc.dart [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import 'dart:math' as math;
 import 'dart:ui' show hashValues, lerpDouble;
 
+import 'package:collection/collection.dart' show maxBy;
```
Comment (Hixie) [inline packages/flutter/lib/src/material/arc.dart]: Can we inline maxBy? We want to remove our dependency on this package if possible.

Comment (jason-simmons) [inline packages/flutter/lib/src/material/arc.dart]: Moved this into foundation/basic_types

File: packages/flutter/lib/src/material/arc.dart [hunk 2]
```diff
@@ -165,14 +166,7 @@ class MaterialRectArcTween extends RectTween {
...
     final Offset centersVector = end.center - begin.center;
-    double maxSupport = 0.0;
-    for (_Diagonal diagonal in _allDiagonals) {
-      final double support = _diagonalSupport(centersVector, diagonal);
-      if (support > maxSupport) {
-        _diagonal = diagonal;
-        maxSupport = support;
-      }
-    }
+    _diagonal = maxBy(_allDiagonals, (_Diagonal d) => _diagonalSupport(centersVector, d));
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/arc.dart]: NICE

**Comment by HansMuller** [comment]
LGTM

**Pull request merged**